src/common/utils/companion.ts

```diff
@@ -0,0 +1,73 @@
+import { PROTOCOL_TOKEN_ADDRESS } from '@common/mocks/tokens';
+import { FullPosition, Permission, Position } from '@types';
+
+export const getCompanionNeedsWithdraw = (position?: Nullable<FullPosition | Position>) => {
```

Comment:
I don't fully understand the name, does this mean something like `check if we need to add withdraw permissions to the companion?` if that is the case, I'd do `doesCompanionNeedWithdrawPermission` or something more verbose.

Reply:
Sure. This function is to check if the companion needs the withdraw permission, it doesnt take into account if the position already has the permission